doc/md/tutorial-todo-gql-field-collection.md

```diff
@@ -0,0 +1,173 @@
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` JOIN (SELECT `todo_parent` FROM `todos` WHERE `id` = ?) AS `t1` ON `todos`.`id` = `t1`.`todo_parent` LIMIT 2
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` JOIN (SELECT `todo_parent` FROM `todos` WHERE `id` = ?) AS `t1` ON `todos`.`id` = `t1`.`todo_parent` LIMIT 2
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` JOIN (SELECT `todo_parent` FROM `todos` WHERE `id` = ?) AS `t1` ON `todos`.`id` = `t1`.`todo_parent` LIMIT 2
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` JOIN (SELECT `todo_parent` FROM `todos` WHERE `id` = ?) AS `t1` ON `todos`.`id` = `t1`.`todo_parent` LIMIT 2
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` JOIN (SELECT `todo_parent` FROM `todos` WHERE `id` = ?) AS `t1` ON `todos`.`id` = `t1`.`todo_parent` LIMIT 2
+```
+
+Let's use Ent for auto-solve our problem. The solution is very simple as all we need to do is to add the following
+`entql` annotations to our edges:
+
+```diff
+func (Todo) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("parent", Todo.Type).
++			Annotations(entgql.Bind()).
+			Unique().
+			From("children").
++			Annotations(entgql.Bind()),
+	}
+}
+```
+
+After adding these annotations, `entgql` will do the binding mentioned in the [section](#ent-solution) above, but it
+also generates edge-resolvers for the nodes under the `edge.go` file as follows:
+
+```go
+func (t *Todo) Children(ctx context.Context) ([]*Todo, error) {
+	result, err := t.Edges.ChildrenOrErr()
+	if IsNotLoaded(err) {
+		result, err = t.QueryChildren().All(ctx)
+	}
+	return result, err
+}
+```
+
+Let's run the code generation again and re-run our GraphQL server:
+
+```console
+go generate ./...
+go run ./cmd/todo
+```
+
+We check the process output again, but this time we'll see that the server executed only 2 queries to the database. 1
+for getting the last 10 todo items, and another for getting the parent-item of each todo-item that was returned in the
+first query.
+
+```sql
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority`, `todos`.`todo_parent` FROM `todos` ORDER BY `id` DESC LIMIT 11
+SELECT DISTINCT `todos`.`id`, `todos`.`text`, `todos`.`created_at`, `todos`.`status`, `todos`.`priority` FROM `todos` WHERE `todos`.`id` IN (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+```
+
+If you're having troubles running this example, go to the [first section](#clone-the-code-optional), clone the code
+and run the example.
+
+---
+
+Please continue to the next section where we explain how to configure our GraphQL mutations to be transactional.
```

Comment:
Well done! By using `entgql.Bind()` in the `ent` schema definition, we were able to greatly improve the efficiency of queries to our application. In the next section, we will learn how to make our GraphQL mutations transactional.